src/utils/statusChecks.ts

```diff
@@ -0,0 +1,62 @@
+import { context, getOctokit } from '@actions/github/lib/github';
+import type { IResult } from '../data';
+import { log } from './action';
+
+interface IContext {
+  owner: string;
+  repo: string;
+  sha: string;
+}
+
+export const createTestStatusCheck = async (
+  token: string,
+  result: IResult,
+  formattedSummary: string,
+  title: string
+): Promise<void> => {
+  const { owner, repo, sha } = getContext();
+
+  if (!token || !owner || !repo || !sha) {
+    log('Failed to create status check - missing required context');
+    return;
+  }
+
+  const octokit = getOctokit(token);
+  const { success } = result;

```

Comment:
since you don't need the whole result here, you can just declare `success: boolean` parameter instead of `result: IResult` in the function declaration. Alternatively you can also declare it like this `{ success }: IResult`